com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateLoop.cs

```diff
@@ -0,0 +1,328 @@
+                        if (indexOfScriptRunDelayedStartupFrame < 0)
+                        {
+                            Debug.LogError($"{nameof(NetworkUpdateLoop)}.{nameof(Initialize)}: Cannot find index of `{nameof(EarlyUpdate.ScriptRunDelayedStartupFrame)}` loop system in `{nameof(EarlyUpdate)}`'s subsystem list!");
+                            return;
+                        }
+
+                        // insert before `EarlyUpdate.ScriptRunDelayedStartupFrame`
+                        subsystems.Insert(indexOfScriptRunDelayedStartupFrame, NetworkEarlyUpdate.CreateLoopSystem());
+                    }
+                    playerLoopSystem.subSystemList = subsystems.ToArray();
+                }
+                else if (playerLoopSystem.type == typeof(FixedUpdate))
+                {
+                    var subsystems = playerLoopSystem.subSystemList.ToList();
+                    {
+                        int indexOfScriptRunBehaviourFixedUpdate = subsystems.FindIndex(s => s.type == typeof(FixedUpdate.ScriptRunBehaviourFixedUpdate));
+                        if (indexOfScriptRunBehaviourFixedUpdate < 0)
+                        {
+                            Debug.LogError($"{nameof(NetworkUpdateLoop)}.{nameof(Initialize)}: Cannot find index of `{nameof(FixedUpdate.ScriptRunBehaviourFixedUpdate)}` loop system in `{nameof(FixedUpdate)}`'s subsystem list!");
+                            return;
+                        }
+
+                        // insert before `FixedUpdate.ScriptRunBehaviourFixedUpdate`
+                        subsystems.Insert(indexOfScriptRunBehaviourFixedUpdate, NetworkFixedUpdate.CreateLoopSystem());
+                    }
+                    playerLoopSystem.subSystemList = subsystems.ToArray();
+                }
+                else if (playerLoopSystem.type == typeof(PreUpdate))
+                {
+                    var subsystems = playerLoopSystem.subSystemList.ToList();
+                    {
+                        int indexOfPhysicsUpdate = subsystems.FindIndex(s => s.type == typeof(PreUpdate.PhysicsUpdate));
+                        if (indexOfPhysicsUpdate < 0)
+                        {
+                            Debug.LogError($"{nameof(NetworkUpdateLoop)}.{nameof(Initialize)}: Cannot find index of `{nameof(PreUpdate.PhysicsUpdate)}` loop system in `{nameof(PreUpdate)}`'s subsystem list!");
+                            return;
+                        }
+
+                        // insert before `PreUpdate.PhysicsUpdate`
+                        subsystems.Insert(indexOfPhysicsUpdate, NetworkPreUpdate.CreateLoopSystem());
+                    }
+                    playerLoopSystem.subSystemList = subsystems.ToArray();
+                }
+                else if (playerLoopSystem.type == typeof(Update))
+                {
+                    var subsystems = playerLoopSystem.subSystemList.ToList();
```

Comment:
Could move this repeated logic to a helper function?
```suggestion
void helper(var subSystemList, var someType)
{
    var subsystems = subSystemList.ToList();
    {
        int indexOfScriptRunBehaviourUpdate = subsystems.FindIndex(s => s.type == typeof(someType));
                        if (indexOfScriptRunBehaviourUpdate < 0)
                        {
     // debug error or probably assert
                            return;
                        }

                        // insert before `Update.ScriptRunBehaviourUpdate`
                        subsystems.Insert(indexOfScriptRunBehaviourUpdate, NetworkUpdate.CreateLoopSystem());
       }
 }
```

Reply:
`PostLateUpdate` adds after not before, also `Initialization` adds to the bottom. we would still keep those two around and can't use helper to work with them :(